### PR TITLE
feat(identity/feature-prune): Add method to remove invalid features

### PR DESCRIPTION
**Pull request by gagantrivedi**

### CODE REVIEW

**Comment by gagantrivedi** [comment]
https://github.com/Flagsmith/flagsmith-engine/blob/main/.flake8 defines a different value for this

File: setup.cfg [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [flake8]
-max-line-length = 88
+max-line-length = 120
```
Comment (matthewelwell) [inline setup.cfg:2]: Is there a good reason to override this here? I think we use the default 88 in our other projects.

File: flag_engine/identities/models.py [hunk 2]
```diff
@@ -39,3 +39,8 @@ def update_traits(self, traits: typing.List[TraitModel]) -> typing.List[TraitMod
 
         self.identity_traits = list(existing_traits.values())
         return self.identity_traits
+
+    def prune_features(self, valid_features: typing.List[str]) -> None:
```
Comment (matthewelwell) [inline flag_engine/identities/models.py:43]: Maybe this argument should be named `valid_feature_names` to be explicit?

Comment (gagantrivedi) [reply flag_engine/identities/models.py]: Yep, makes sense

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**